### ISSUE TITLE
[Enhancement] Friendlier JSON in Preview mode

### ISSUE THREAD

**Issue by jvukovich** [Kong/insomnia#1556]
It looks like Insomnia always escapes the forward slash:
`\/`

Can this be an option in the response window, or the IDE as global option?

This bit us, as a very long encoded value was behaving one way in Insomnia (escaping the forward slash), but was different in another REST client (Postman: it doesn't escape forward slashes).

Thanks!

**Comment by gschier** [comment]
Where are you seeing the forward slash being escaped?

**Comment by jvukovich** [comment]
In the response. Thanks!

**Comment by richardson-trevor** [comment]
Just trying Insomnia for the first time. Gorgeous -- nice work! We're having this issue as well. We return links in our payload in relative form (no scheme or authority). All forward slashes are rendered as escaped making them not clickable.

**Comment by vavra7** [comment]
The same issue for me. Option for disabling of escaping all forward slashes would be very nice.

**Comment by gschier** [comment]
I think having the Preview show more friendly JSON could be a good idea

**Comment by pedzed** [comment]
Any progress on this?

**Comment by K2ouMais** [comment]
Hello everyone, will this be fixed?

I am still getting escape URL/links.

Will it be possible to not show these escaped?

**Comment by K2ouMais** [comment]
Almost a year and no progress on this ?

**Comment by nijikokun** [comment]
Adding label for investigation / verification after attempting to replicate and being unable to do so.

Can someone who is encountering this document the steps to replicate along with the request / response body and headers used to replicate - preferably with screenshots. 

It seems like an edge-case or misconfigured settings / content-types, see screenshot below where my slashes are not escaped:

<img width="1020" alt="Screen Shot 2021-03-24 at 12 45 18 PM" src="https://user-images.githubusercontent.com/198276/112374003-d28b5480-8c9e-11eb-8829-68e8f6938eac.png">

Pasting from the preview pane also does not result in slashes:

```"prop": "hello/world/there"```

**Comment by K2ouMais** [comment]
There is no misconfiguration at all.
I even have installed Insomnia on a brand new PC and without have setting anything URLs are all escaped.
Do a request to an API that is reponding an URL and you will see it escaped. Simple as that.
The content-type I was using was a simple application/json and probably the most common out there.

**Comment by nijikokun** [comment]
@K2ouMais I hear you but without proper steps to replicate and exact information it makes it difficult to work on. What version of Insomnia are you running and what is the url, request body, and headers you are requesting with, as I cannot replicate the issue.

**Comment by K2ouMais** [comment]
Ok, it is really late here. I will try to give you a proper example tomorrow.

**Comment by dimitropoulos** [comment]
maybe http://mockbin.org/ will be useful in simulating such a response

**Comment by K2ouMais** [comment]
Here is a good example:
GET https://api.gleif.org/api/v1/entity-legal-forms

Escaped URLs:
![EscapedLinks](https://user-images.githubusercontent.com/52510455/112431797-45004100-8d40-11eb-9ff0-df356fa70124.png)

Headers:
![Headers](https://user-images.githubusercontent.com/52510455/112432167-c657d380-8d40-11eb-8941-466bcc0a9e51.png)

Preferences using the newest Insomnia version:
![Preferences](https://user-images.githubusercontent.com/52510455/112431896-65300000-8d40-11eb-9a12-67d814ea16dc.png)

This API has probably a Laravel Backend.

I am seeing the same behaviour on my API with a Laravel Backend.

I dont know if this could be the problem.

If I use the exact same example in Postman, every link seems to be right.

**Comment by K2ouMais** [comment]
@dimitropoulos @nijikokun 
I have inserted a proper API where the URLs are escaped.
Could you please take a look at it?

Thank you

**Comment by fcoriverah** [comment]
Having the same issue :( still no fix?

**Comment by K2ouMais** [comment]
Nope. They wanted me to show them the problem and since then nothing.

It has been a month.

**Comment by fcoriverah** [comment]
> Nope. They wanted me to show them the problem and since then nothing.
> 
> It has been a month.

Sad, i really like Insomnia over Postman but this...

**Comment by rmsohaila** [comment]
I've the same issue. Postman works fine for same API. Insomnia please fix it.

**Comment by sgostanyan** [comment]
Same here, spent several hours to undestand why the token I request is not valid with  Insomnia but it is with Postman. It was just because forward slash is escaped :

![sc1](https://user-images.githubusercontent.com/23078069/120808905-d0206300-c549-11eb-91e1-53db2edc7827.png)

**Comment by dimitropoulos** [comment]
@sgostanyan very sorry to hear that you lost time due to this.  I've asked internally if we can get this prioritized in some way.

@K2ouMais sorry on the dealyed response.  A bot to mark issues got enabled which flooded my inbox with 600+ issues from insomnia and your response here got lost in the shuffle.

I think I understand what's going on.

So first off, the response you gave includes the escapes

Here's an example with just curl
![Screenshot_20210604_100139](https://user-images.githubusercontent.com/15232461/120813441-e1eb1180-c51b-11eb-8f1e-4ffed1196aee.png)

Here's an example from chrome:
![Screenshot_20210604_100222](https://user-images.githubusercontent.com/15232461/120813529-f5967800-c51b-11eb-8924-6360936efb77.png)

And, indeed, here's an example from Postman:
![Screenshot_20210604_100255](https://user-images.githubusercontent.com/15232461/120813600-09da7500-c51c-11eb-9d76-5d16de6a92da.png)

The difference here is that in our "Preview" mode, which is sorta like a pretty print, we use the JSON.stringify behavior.

For example,
![Screenshot_20210604_101209](https://user-images.githubusercontent.com/15232461/120814898-53778f80-c51d-11eb-93ad-139158153454.png)

For this behavior to change, we'd need to update something around here: https://github.com/Kong/insomnia/blob/develop/packages/insomnia-app/app/ui/components/codemirror/code-editor.tsx#L634

I made a mockbin we can use for this: http://mockbin.org/bin/6f5df8ac-8fa4-4aa0-8961-d01ed9f29d7a

![Screenshot_20210604_105732](https://user-images.githubusercontent.com/15232461/120821646-aa806300-c523-11eb-8038-00aa0651d010.png)

**Comment by jvukovich** [comment]
Any progress on this? I could take a stab at a PR, but, not being at all familiar with the codebase, I'd probably do it wrong.

**Comment by Harrylepap** [comment]
Still no progress ?

**Comment by massimocode** [comment]
Is there any chance this issue can be prioritised? I'm really not sure why forward slashes in network responses are being escaped in the Preview tab.

**Comment by rusagaib** [comment]
i think someone has addressed this kind issue long ago and it has been closed 🤔 pls check this @jvukovich [link-issue](https://github.com/Kong/insomnia/issues/315#issuecomment-308937313)

**Comment by jvukovich** [comment]
Thanks, I wasn't aware of the other closed issue.

Well, there looks like there are quite a few recent follow-ups on that closed issue, too, particularly mrjimhubbard's suggestion:
![image](https://github.com/Kong/insomnia/assets/6004664/a42c756f-236d-4014-91b6-bad01d66d6db)

Should we not leave this issue open if enough people are rallying around making this clearer/easier/user-friendly?

**Comment by jvukovich** [comment]
👍❤️❤️❤️❤️👍
